extension/data/background/handlers/notifications.js

```diff
@@ -76,14 +118,22 @@ async function sendPageNotification ({title, body, url, modHash, markreadid}) {
     return notificationID;
 }
 
+// Handle notification clearing
+browser.alarms.onAlarm.addListener(alarmInfo => {
+    const name = alarmInfo.name;
+    if (name.startsWith('tb-notification-')) {
+        const notificationID = name.replace('tb-notification-', '');
+        clearNotification(notificationID);
+    }
+});
+
 // Handle notification creation
 messageHandlers.set('tb-notification', request => {
-    const notificationTimeout = 6000;
     const sendNotification = request.native ? sendNativeNotification : sendPageNotification;
     sendNotification(request.details).then(id => {
-        setTimeout(() => {
-            clearNotification(id);
-        }, notificationTimeout);
+        browser.alarms.create(`tb-notification-${id}`, {
+            delayInMinutes: 1,
```

Comment:
1 minute is significantly more than the 6 seconds this was set to before... according to MDN, this accepts a float, so we can do:

```suggestion
            delayInMinutes: 1 / 60 * 6,
```

or alternatively use `when` and specify the absolute time:

```suggestion
            when: Date.now() + 6000,
```

Reply:
this api is dumb of course you can't do that